### ISSUE TITLE
Implement binary tree for pavings

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/IntervalConstraintProgramming.jl#3]

**Comment by dpsanders** [comment]
https://en.wikipedia.org/wiki/K-d_tree

Ray tracing uses this and has a way to find the neighbouring box. (Thanks to @c42f)

**Comment by c42f** [comment]
Yes, to me it looks like you're implicitly creating a kdtree during your splitting process.  Whether that's actually useful is something I don't know - ray tracing traversal is a somewhat different use case :-)

**Comment by dpsanders** [comment]
The splitting process actually creates a binary tree, since each interval is bisected only once (currently the side of maximum length is split).

**Comment by dpsanders** [comment]
In progress in the `paving_as_tree` branch.
